java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -211,4 +216,65 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
       addMetadata(pwsResult);
     }
   }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls, double minRank) {
+    // Get all valid PwPairs.
+    List<PwPair> allPwPairs = new ArrayList<>();
+    for (UrlDevice urlDevice : mDeviceIdToUrlDeviceMap.values()) {
+      PwsResult pwsResult = mBroadcastUrlToPwsResultMap.get(urlDevice.getUrl());
+      if (pwsResult != null) {
+        allPwPairs.add(new PwPair(urlDevice, pwsResult));
+      }
+    }
+
+    // Sort the list in descending order.
+    Collections.sort(allPwPairs);
+    Collections.reverse(allPwPairs);
+
+    // Filter the list.
+    List<PwPair> ret = new ArrayList<>();
+    Set<String> siteUrls = new HashSet<>();
+    for (PwPair pwPair : allPwPairs) {
+      String siteUrl = pwPair.getPwsResult().getSiteUrl();
+      if (pwPair.getRank() >= minRank
+          && !dedupSiteUrls && !siteUrls.contains(siteUrl)) {
+        siteUrls.add(siteUrl);
+        ret.add(pwPair);
+      }
+    }
+
+    return ret;
+  }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls) {
+    return getPwPairsSortedByRank(dedupSiteUrls, Double.NEGATIVE_INFINITY);
+  }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(double minRank) {
```

Comment:
The rank is implemented by custom UrlDevice definitions, so even in the case of an ordered list, one would be created.

We can drop the overloads though